### PR TITLE
ble service: Battery Service

### PR DESCRIPTION
**Pull request by sdalu**
Main characteristics for the Battery Service

### CODE REVIEW

File: net/nimble/host/services/bas/syscfg.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+syscfg.defs:
+    BLE_SVC_BAS_BATTERY_LEVEL_PERM:
+        description: >
+            Defines permissions for writing "Battery Level" characteristics. Can
+            be zero to allow write without extra permissions or combination of:
```
Comment (jacobrosenthal) [inline net/nimble/host/services/bas/syscfg.yml]: Defines permissions for *reading "Battery Level"
and
Can be zero to allow *read without extra permissions or combination of

And maybe BLE_SVC_BAS_BATTERY_LEVEL_PERM should be BLE_SVC_BAS_BATTERY_LEVEL_READ_PERM but I dont particularly care either way.

Comment (andrzej-kaczmarek) [reply net/nimble/host/services/bas/syscfg.yml]: +1 for READ_PERM

File: net/nimble/host/services/bas/syscfg.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                BLE_GATT_CHR_F_READ_ENC
+                BLE_GATT_CHR_F_READ_AUTHEN
+                BLE_GATT_CHR_F_READ_AUTHOR
+        value: 0
+    BLE_SVC_BAS_BATTERY_LEVEL_NOTIFY:
+        description: >
+            Set to BLE_GATT_CHR_F_NOTIFY to support notification,
+            or 0 to disable it.
+        value: BLE_GATT_CHR_F_NOTIFY
```
Comment (andrzej-kaczmarek) [inline net/nimble/host/services/bas/syscfg.yml]: this should be a simple 0/1 value - it's not a permissions mask, just simple bool to either enable or disable notify prop

File: net/nimble/host/services/bas/src/ble_svc_bas.c [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+ * If greater than 100, it will be silently set to 100.
+ */
+int
+ble_svc_bas_battery_level_set(uint8_t level) {
+    if (level > 100)
+	level = 100;
+    ble_svc_bas_battery_level = level;
+#if MYNEWT_VAL(BLE_SVC_BAS_BATTERY_LEVEL_NOTIFY) > 0
+    return ble_gattc_notify(ble_svc_bas_conn_handle, 
+                            ble_svc_bas_battery_handle);
```
Comment (andrzej-kaczmarek) [inline net/nimble/host/services/bas/src/ble_svc_bas.c]: we probably should not notify if value did not change, i.e. if ble_svc_bas_battery_level == level

File: net/nimble/host/services/bas/syscfg.yml (hunk 2 above)
Comment (jacobrosenthal) [reply net/nimble/host/services/bas/syscfg.yml]: or I suppose the spec allows to permission these? So you could make it a permissions mask?

Comment (andrzej-kaczmarek) [reply net/nimble/host/services/bas/syscfg.yml]: Based on some informations scattered in different places of spec, notifications should most likely follow permissions of CCCD. However, NimBLE does not allow to set them and obviously also does not enforce them so there's nothing to set here actually :)

I'd leave this as NOTIFY_ENABLE 0/1 syscfg for now and I'll look into adding appropriate support in Mynewt - we can then add NOTIFY_PERM syscfg and update code accordingly.

**Comment by sdalu** [comment]
See: https://github.com/apache/mynewt-nimble/pull/13

**Pull request closed without merging**